### ISSUE TITLE
Adding HCA as datasource to Zooma 

### ISSUE THREAD

**Issue by clairerye** [ebi-ait/dcp-ingest-central#230]
Adding HCA as datasource to Zooma from ticket: https://app.zenhub.com/workspaces/operations-5fa2d8f2df78bb000f7fb2b5/issues/ebi-ait/hca-ebi-wrangler-central/104
Description of the task:

Use the attached csv template for loading annotations into Zooma.

You can find a description of the values that need be supplied in the csv file at https://www.ebi.ac.uk/spot/zooma/docs/getting-started.html under the section "Adding New Mappings to Zooma".

Once you have a populated csv file, tell the SPOT team the URL from where they can load this file from.
This URL needs to be constant and cannot change with different releases from HCA. Many of Zooma datasources are loaded from ftp://ftp.ebi.ac.uk/pub/databases/spot/zooma/data....
So for example ftp://ftp.ebi.ac.uk/pub/databases/spot/zooma/data/annotations/hca/latest/hca_annotations.csv may be viable option.

Once we have provided the SPOT team with the information in (3), they will load it into their test environment to give us an opportunity to review and spot any possible mistakes.

Once you give the SPOT team the go-ahead on 4, they will make it available on prod and notify us.

I think we need to plan the update-upload-testing cycle as a recurring process.

**Acceptance criteria for the task:**

-  complete the first upload, testing and production cycle
-  establish process for update-upload-testing cycles
- Adding HCA as datasource to Zooma hasno Dependencies

**rays22 added taskoperations labels 8 months ago**

**rays22
commented on Jul 24, 2020**
zooma-template.csv:

> ANNOTATION_URI,ANNOTATION_ID,STUDY_URI,STUDY,STUDY_ID,STUDY_TYPE,BIOENTITY_URI,BIOENTITY,BIOENTITY_TYPE_URI,BIOENTITY_TYPE_NAME,BIOENTITY_ID,PROPERTY_URI,PROPERTY_ID,PROPERTY_TYPE,PROPERTY_VALUE,SEMANTIC_TAG,ANNOTATOR,ANNOTATION_DATE
> See 5 older events
> Automatically changed the pipeline from New Issues to To do in Marions backlog based on Workflows 4 months ago
> Automatically changed the pipeline from New issues to Queued for wrangling in Dataset wrangling status based on Workflows 4 months ago

**lauraclarke
commented on Nov 18, 2020**

> @rays22 can you add details here of what you need @aaclan-ebi's help with

rays22 
commented on Nov 19, 2020
> I need help on how I can get all the ontology related metadata field values for all the datasets that we have wrangled in bulk.
> 

> *.text *.ontology *.ontology_label
> I have tested it and I can get that information for each dataset by using the Ingest API if I know the submissionEnvelope and query the appropriate endpoints:

```
"submissionEnvelope" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX",
  "title" : "A single submission envelope"
},
"biomaterials" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/biomaterials"
},
"processes" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/processes"
},
"files" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/files",
  "title" : "Access or create, within a submission envelope, a new assay"
},
"projects" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/projects",
  "title" : "Access or create projects. Creation can only be done inside a submission envelope"
},
"protocols" : {
  "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/protocols",
  "title" : "Access or create protocols"
```

> The problem is that I have not found a trivial way to get a list of all the submissionEnvelope values for all the wrangled datasets. If I go to Ingest UI and log in, then I can open each project one-by-one and then I can find the submissionEnvelope for the latest submission for that project.
> I am not sure how long it would take for me to find all the required submissionEnvelope IDs for all the wrangled datasets this way.
> 
> I would like to ask @aaclan-ebi if there is a better and faster way to get the submissionEnvelope IDs for all the wrangled projects/latest submissions.
> 
> Alternatively, there may be completely different and better ways to get the data in bulk from the ontology related fields, e.g Mongo DB? Or Terra Data Repo?
> Actually I could write more advanced SQL queries more quickly than scripts to parse JSON responses...
> 
> Shall I make another ticket about getting the bulk data? If yes, would it be a development ticket or operations?

**mshadbolt
commented on Jan 4, 2021**

> I got a list of submission envelope IDs and am using this script to get all the relevant ontology mappings : https://github.com/HumanCellAtlas/hca-data-wrangling/blob/master/src/ontology_mappings_extractor.py
> 
> @rays22 are you aware of whether the csv that we provide should be non-duplicated, i.e. each match between a provided text field and an ontology occurs only once, or should appear as many times as in our data so that it provides more weight to particular annotations?

**rays22
commented on Jan 4, 2021**

> Thanks @mshadbolt .
> I think the text label -- preferred ontology term pairs should be unique in the csv.
> As I understand it, a particular mapping will get more weight in zooma if it appears in multiple independent mappings, not when it appears in multiple times in one source.
> I think we will be able to select the HCA-specific mappings, so those HCA-preferred mappings should appear as top hits.

**Comment by mshadbolt** [comment]
I have made progress on this ticket with the ontology team. A test instance can be used here when logged in to the EBI VPN:

> We have been to upload it to Zooma. When you are on the VPN you can access it at: http://snarf.ebi.ac.uk:8580/spot/zooma/
> 
> As a simple example I tested annotating: blood    organ
> 
> Note that there is a TAB between blood and organ

Tasks that remain are:
- come up with a documented process to automatically add curations to the existing zooma config file at regular intervals
- Add existing cell type annotations to the zooma config file

**Comment by mshadbolt** [comment]
There is now a pending PR for an initial version of this process https://github.com/ebi-ait/hca-ebi-wrangler-central/pull/326

It will need to iterated and improved on once we become more familiar with how Zooma works and how it can best help with our ontology curation work
